app/assets/javascripts/controllers/report_data_controller.js

```diff
@@ -0,0 +1,357 @@
+  */
+  ReportDataController.prototype.onItemSelect = function(item, isSelected) {
+    var selectedItem = _.find(this.gtlData.rows, {id: item.id});
+    if (selectedItem) {
+      selectedItem.checked = isSelected;
+      selectedItem.selected = isSelected;
+      this.$window.sendDataWithRx({rowSelect: selectedItem});
+      if (isSelected) {
+        ManageIQ.gridChecks.push(item.id);
+      } else {
+        var index = ManageIQ.gridChecks.indexOf(item.id);
+        index !== -1 && ManageIQ.gridChecks.splice(index, 1);
+      }
+    }
+  };
+
+  ReportDataController.prototype.initObjects = function(initObject) {
+    this.gtlData = { cols: [], rows: [] };
+    this.initObject = initObject;
+    if (this.initObject.showUrl === '') {
+      this.initObject.showUrl = '/' + ManageIQ.controller;
+      if (this.initObject.isExplorer) {
+        this.initObject.showUrl += '/x_show/';
+      } else {
+        this.initObject.showUrl += '/show/';
+      }
+    }
+    this.gtlType = initObject.gtlType || 'grid';
+    this.settings.isLoading = true;
+    ManageIQ.gridChecks = [];
+    this.$window.sendDataWithRx({setCount: 0});
+  };
+
+  /**
+  * Method for initializing controller. Good for bootstraping controller after loading items. This method will call
+  * getData for fetching data for current state. After these data were fetched, sorting items and filtering them takes
+  * place.
+  * @param {Object} initObject this object will hold all information about current state.
+  * ```
+  *   initObject: {
+  *     modelName: string,
+  *     gtlType: string,
+  *     activeTree: string,
+  *     currId: string,
+  *     isExplorer: Boolean
+  *   }
+  * ```
+  * @returns {Object} promise of fetched data.
+  */
+  ReportDataController.prototype.initController = function(initObject) {
+    this.$window.ManageIQ.gtl = this.$window.ManageIQ.gtl || {};
+    this.$window.ManageIQ.gtl.loading = true;
+    initObject.modelName = decodeURIComponent(initObject.modelName);
+    this.initObjects(initObject);
+    this.setExtraClasses(initObject.gtlType);
+    this.showMessage = this.$location.search().flash_msg;
+    return this.getData(initObject.modelName,
+                        initObject.activeTree,
+                        initObject.currId,
+                        initObject.isExplorer,
+                        this.settings,
+                        initObject.records)
+      .then(function(data) {
+        this.settings.hideSelect = initObject.hideSelect;
+        var start = (this.settings.current - 1) * this.settings.perpage;
+        this.setPaging(start, this.settings.perpage);
+        var sortId = _.findIndex(this.gtlData.cols, {col_idx: parseInt(this.settings.sort_col, 10)});
+        if (sortId !== -1) {
+          this.setSort(sortId, this.settings.sort_dir === 'ASC');
+        }
+        this.setDefaults();
+        this.movePagination();
+        this.$timeout(function() {
+          this.$window.ManageIQ.gtl.loading = false;
+        });
```

Comment:
missing `.bind(this)`.

Otherwise, `this.$window` is undefined and causes 

```
TypeError: Cannot read property 'ManageIQ' of undefined
    at report_data_controller.self-e8b2f4f5159331da9e269ab771047c87f3ee0694675b3a6d367f421c21a71a5b.js:251
    at angular.self-a4ec94b978004b45127819f4f8dda67f742334b1e952c2dc446a327714b45fb5.js:19943
    at completeOutstandingRequest (angular.self-a4ec94b978004b45127819f4f8dda67f742334b1e952c2dc446a327714b45fb5.js:6046)
    at angular.self-a4ec94b978004b45127819f4f8dda67f742334b1e952c2dc446a327714b45fb5.js:6325
```